eHealthAfrica/couchdb-user-management-app
# Men 379

Summary of this PR:

- Introduces a middleware to pass filter details to the User model
- Updates User model fetch and search code to make use of the filter details passed to filter the data sent to the user. In the case of the fetch method, it was refactored to  otherwise rely on the regular query mechanism which is more efficient
- Updates the test script  and sample ums file
-Updates approach to dependency fetching, previous implementation had a flaw of not updating data.



How to test this PR:
- Test with LMIS-Dashboard MEN-379 branch.
- Ensure you have dashboard users that are active and admins that have been assigned to different locations.
-Using each of the users above, log in and navigate to the UMS and observe the users displayed.
